### PR TITLE
Add juicefsruntime dataload function

### PR DESCRIPTION
**Pull request by ldd91**
### Ⅰ. Describe what this PR does
Add juicefs dataload function

### Ⅴ. Special notes for reviews

### CODE REVIEW

File: charts/fluid-dataloader/juicefs/values.yaml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+      replicas: 1
+      fluidNative: false
+
+  # Required
+  # Description: the image that the DataLoad job uses
+  image: #<jindo-image>
+
+podNames:
+  - ""
+namespace: ""
```
Comment (cheyang) [inline charts/fluid-dataloader/juicefs/values.yaml:34]: Add return of line.

File: pkg/ddc/juicefs/cacheinfo_parser.go [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+	"github.com/fluid-cloudnative/fluid/pkg/utils/kubeclient"
+	"github.com/pkg/errors"
+	"gopkg.in/yaml.v2"
+	v1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// G
+func GetCacheInfoFromConfigmap(client client.Client, name string, namespace string) (cacheinfo map[string]string, err error) {
```
Comment (cheyang) [inline pkg/ddc/juicefs/cacheinfo_parser.go:30]: Do you want to write comments for function GetCacheInfoFromConfigmap.

File: charts/fluid/fluid/templates/role/dataset/rbac.yaml [hunk 3]
```diff
@@ -49,9 +49,12 @@ rules:
   - apiGroups:
       - ""
     resources:
-      - events
```
Comment (cheyang) [inline charts/fluid/fluid/templates/role/dataset/rbac.yaml:52]: Why is `events` missing?

File: charts/fluid/fluid/templates/role/dataset/rbac.yaml [hunk 4]
```diff
@@ -61,6 +64,16 @@ rules:
       - statefulsets/status
     verbs:
       - '*'
+  - apiGroups:
```
Comment (cheyang) [inline charts/fluid/fluid/templates/role/dataset/rbac.yaml:67]: Is necessary to enlarge the RBAC authorization of dataset controller?

File: charts/fluid/fluid/templates/role/dataset/rbac.yaml (hunk 3 above)
Comment (zwwhdls) [reply charts/fluid/fluid/templates/role/dataset/rbac.yaml:52]: `events` bas been involved in line22.

File: pkg/ddc/juicefs/cacheinfo_parser.go (hunk 2 above)
Comment (zwwhdls) [reply pkg/ddc/juicefs/cacheinfo_parser.go:30]: done

File: charts/fluid-dataloader/juicefs/templates/configmap.yaml [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+        podNames="$POD_NAMES"
+        podNames=(${podNames//:/ })
+
+        ns="$POD_NAMESPACE"
+        for((i=0;i<${#podNames[@]};i++)) do
+          local pod="${podNames[i]}"
+
+          for((j=0;j<${#paths[@]};j++)) do
+            echo -e "juicefs warmup on $pod ${paths[j]} starts"
+            /usr/local/bin/kubectl -n $ns exec -it $pod -- $COMMAND
```
Comment (cheyang) [inline charts/fluid-dataloader/juicefs/templates/configmap.yaml:26]: Will  you add `timeout` to avoid the job hang?

**Comment by cheyang** [comment]
Please fix no check issue. thanks.

```
pkg/ddc/juicefs/data_load.go:208:25: Error return value of `fileUtils.UnMount` is not checked (errcheck)
	defer fileUtils.UnMount(mountPath)
```

File: charts/fluid-dataloader/juicefs/templates/configmap.yaml [hunk 6]
```diff
@@ -23,9 +23,9 @@ data:
 
           for((j=0;j<${#paths[@]};j++)) do
             echo -e "juicefs warmup on $pod ${paths[j]} starts"
-            /usr/local/bin/kubectl -n $ns exec -it $pod -- $COMMAND
-            /usr/local/bin/kubectl -n $ns exec -it $pod -- juicefs warmup $MOUNTPATH${paths[j]}
-            /usr/local/bin/kubectl -n $ns exec -it $pod -- umount $MOUNTPATH
+            /usr/local/bin/kubectl -n $ns exec -it $pod --request-timeout=10s -- $COMMAND
+            /usr/local/bin/kubectl -n $ns exec -it $pod --request-timeout=1m -- juicefs warmup $MOUNTPATH${paths[j]}
```
Comment (cheyang) [inline charts/fluid-dataloader/juicefs/templates/configmap.yaml:27]: I think the timeout of warmup behavior should be configurable. By the way, is 1Minute enough for juicefs warm scenario?

File: charts/juicefs/templates/worker/statefuleset.yaml [hunk 7]
```diff
@@ -58,25 +58,71 @@ spec:
...
         - name: cache-dir
           hostPath:
             path: {{ .Values.worker.cacheDir }}
             type: DirectoryOrCreate
-      {{- end }}
+        {{- end }}
+        - name: script
+          configMap:
+            name: {{ template "juicefs.fullname" . }}-script
+            defaultMode: 0777
```
Comment (cheyang) [inline charts/juicefs/templates/worker/statefuleset.yaml:128]: Why not setting `0755`?

File: charts/fluid/fluid/templates/role/dataset/rbac.yaml (hunk 4 above)
Comment (zwwhdls) [reply charts/fluid/fluid/templates/role/dataset/rbac.yaml]: done

File: charts/fluid-dataloader/juicefs/templates/configmap.yaml (hunk 5 above)
Comment (zwwhdls) [reply charts/fluid-dataloader/juicefs/templates/configmap.yaml]: done

File: charts/fluid-dataloader/juicefs/templates/configmap.yaml (hunk 6 above)
Comment (zwwhdls) [reply charts/fluid-dataloader/juicefs/templates/configmap.yaml]: done

File: charts/juicefs/templates/worker/statefuleset.yaml (hunk 7 above)
Comment (zwwhdls) [reply charts/juicefs/templates/worker/statefuleset.yaml]: `0755` is ok, done

**Review by cheyang — APPROVED** [review]

**Pull request merged**